kerastuner/engine/hypermodel.py

```diff
@@ -19,13 +19,41 @@
 
 
 class HyperModel(object):
+    """Defines a searchable space of Models and builds Models from this space.
+
+    Attributes:
+        name: The name of this HyperModel.
+        tunable: Whether the hyperparameters defined in this hypermodel
+          should be added to search space. If `False`, either the search
+          space for these parameters must be defined in advance, or the
+          default values will be used.
+    """
 
     def __init__(self, name=None, tunable=True):
         self.name = name
+        self.tunable = tunable
+
+        self._build = self.build
+        self.build = self._tunable_aware_build
 
     def build(self, hp):
+        """Builds a model.
+
+        # Arguments:
+            hp: A `HyperParameters` instance.
+
+        # Returns:
+            A model instance.
+        """
         raise NotImplementedError
 
+    def _tunable_aware_build(self, hp):
```

Comment:
We may want to use a more generic name, such as `_build_wrapper` or `_build_impl`, as we may need to add more functionality to it later.

Reply:
Done